# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/BytecodeControlStructures.h
# Commit: 7a3a4207c55b
# Full Hash: 7a3a4207c55b27d6a73783c0021cfed2368bdbb8
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-06 21:48:33
# Regressor Bug: 1601897
# File Overlap Count: 3
# Description:
#   Bug 1601897 part 2 - Move patchBreaks and addTryNote calls to LoopControl::emitLoopEnd. r=arai
#   
#   This ensures all loops handle this the same way.
#   
#   For JSTRY_FOR_IN some places asserted the try note ends at the
# ==============================================================================

diff -r 7a9ab8423e85 -r 7a3a4207c55b js/src/frontend/BytecodeControlStructures.h
--- a/js/src/frontend/BytecodeControlStructures.h	Fri Dec 06 11:08:02 2019 +0000
+++ b/js/src/frontend/BytecodeControlStructures.h	Fri Dec 06 11:08:04 2019 +0000
@@ -142,7 +142,8 @@
   MOZ_MUST_USE bool emitLoopHead(BytecodeEmitter* bce,
                                  const mozilla::Maybe<uint32_t>& nextPos);
 
-  MOZ_MUST_USE bool emitLoopEnd(BytecodeEmitter* bce, JSOp op);
+  MOZ_MUST_USE bool emitLoopEnd(BytecodeEmitter* bce, JSOp op,
+                                JSTryNoteKind tryNoteKind);
 };
 template <>
 inline bool NestableControl::is<LoopControl>() const {